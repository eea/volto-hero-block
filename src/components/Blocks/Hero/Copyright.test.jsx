import React from 'react';
import renderer from 'react-test-renderer';
import Copyright from './Copyright';
import { Icon } from 'semantic-ui-react';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import config from '@plone/volto/registry';

const mockStore = configureStore();

test('renders copyright', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const copyrightPrefix =
    config.blocks.blocksConfig.hero?.copyrightPrefix || '';

  const component = renderer.create(
    <Provider store={store}>
      <Copyright copyrightPosition={'left'}>
        <Copyright.Prefix>{copyrightPrefix}</Copyright.Prefix>
        <Copyright.Icon>
          <Icon className={'copyright outline'} />
        </Copyright.Icon>
        <Copyright.Text>Copyright Text Test</Copyright.Text>
      </Copyright>
    </Provider>,
  );

  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
