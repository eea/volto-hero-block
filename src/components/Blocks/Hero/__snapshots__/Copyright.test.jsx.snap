// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders copyright 1`] = `
<div
  className="eea copyright left"
>
  <div
    className="wrapper"
  >
    
    <span
      className="icon-wrapper"
    >
      <i
        aria-hidden="true"
        className="icon copyright outline"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon-content"
    >
      Copyright Text Test
    </span>
  </div>
</div>
`;
